packages/core/types/theming.d.ts

```diff
@@ -21,26 +12,13 @@ export interface ThemeProvider<Theme extends {} = any> {
   (props: ThemeProviderProps<Theme>): React.ReactElement
 }
 
-export type withTheme<Theme extends {} = any> = <
```

Comment:
I've removed it as part of this PR - but the decision about it is not final. It seems like in hooks world this is not needed. I know class components can't use hooks but they can do smth like that:
```js
// this example is not generic, but also `withTheme` tends to be used sparingly, so this might not be a problem
const WithTheme = props => {
  const theme = useTheme()
  return <SomeClass theme={theme} {...props}/>
}
```

So I don't know if we should keep builtin `withTheme`.

Reply:
We have pretty much migrated to hooks, but it has taken time. I think at a minimum you should add in the upgrade notes a snippet to polyfill it.

Reply:
Sure, a migration path for each change is going to be documented. I think this one even could be automated with codemods, just need to consider people wrapping components with multiple HOCs at once.